Monika-After-Story/MonikaModDev
# Player bday update

This helps the player bday event be more sustainable, offering a little variety and the ability to easily select a new unseen poem every year. The changes to the autoload only affect the normal path where we would have gotten to the open-door surprise greet before.

## Adds

- `persistent._mas_player_bday_saw_surprise` a new var that stores if we got the surprise party variant of the birthday event in a given year.

- `mas_player_bday_greet` a new greet that, depending on circumstances, is a fake open-door greet where Monika already has everything set up, or a normal greet that starts the bday party.

- `getRandomPoem` a `mas_poems` func that gets a random poem of a specified category. can be used to get an unseen poem to facilitate the player getting a new poem they have yet to receive.

## Testing
- verify on your first birthday party with Monika (or if you've only gotten the party prior via `mas_player_bday_ret_on_bday` or `mas_player_bday_no_restart`) you get the special open door greet this year if you load in normally on your bday.

- verify if you ruin the surprise (open the door before knocking), next year, if you load in normally, you do not get the open door greet, but instead get the `else` path of `mas_player_bday_greet`.

- verify the bday poems don't repeat until you've seen them all. may want to add a few dummy poems and just `call mas_showpoem(mas_poems.getRandomPoem("pbday"))` via console to confirm this without time travelling multiple times.

- verify the few dialogue alternatives added make sense based on if it's your first or subsequent bday party spent with Monika.
